tests/distributions/test_sine_bivariate_von_mises.py

```diff
@@ -0,0 +1,131 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+import torch
+from scipy.special import binom, iv
+from torch import tensor
+from torch.distributions import Beta, HalfNormal, VonMises
+
+import pyro
+from pyro.distributions import Geometric, constraints
+from pyro.distributions.sine_bivariate_von_mises import SineBivariateVonMises, log_I1
+from pyro.infer import SVI, Trace_ELBO
+from tests.common import assert_equal
+
+
+def _unnorm_log_prob(value, loc1, loc2, conc1, conc2, corr):
+    if len(value.shape) == 1:
+        phi_val = value[0]
+        psi_val = value[1]
+    else:
+        phi_val = value[..., 0]
+        psi_val = value[..., 1]
+    return (conc1 * torch.cos(phi_val - loc1) + conc2 * torch.cos(psi_val - loc2) +
+            corr * torch.sin(phi_val - loc1) * torch.sin(psi_val - loc2))
+
+
+@pytest.mark.parametrize('n', [0, 1, 10, 20])
+def test_log_binomial(n):
+    comp = SineBivariateVonMises._lbinoms(tensor(n))
+    act = tensor([binom(2 * i, i) for i in range(n)]).log()
+    assert_equal(act, comp)
+
+
+def test_log_I1_shapes():
+    assert_equal(log_I1(10, tensor(.6)).shape, torch.Size([11, 1]))
+    assert_equal(log_I1(10, tensor([.6])).shape, torch.Size([11, 1]))
+    assert_equal(log_I1(10, tensor([[.6]])).shape, torch.Size([11, 1, 1]))
+    assert_equal(log_I1(10, tensor([.6, .2])).shape, torch.Size([11, 2]))
+    assert_equal(log_I1(0, tensor(.6)).shape, torch.Size((1, 1)))
+
+
+@pytest.mark.parametrize('order', [0, 1, 5, 10, 20])
+def test_log_I1(order):
+    expected = torch.tensor([iv(i, tensor((.6,)).numpy()) for i in range(order + 1)]).log()
+    actual = log_I1(order, tensor([.6]))
+    assert_equal(actual, expected)
```

Comment:
If you move `log_I1` to pyro.ops.special, you could move these tests to tests/ops/test_special.py.

I'd also recommend a small test checking values against scipy.

Reply:
I'll move this to `test_special.py`. By checking values against SciPy, do you mean checking more values than just .6 as `test_log_I1` does?

Reply:
Oh, indeed I missed your test.  Yes, I'd test against a range of values e.g.
```py
torch.tensor([0.01, 0.1, 0.6, 1.0, 10.0, 100.0])
```

Reply:
That makes sense; I'll add this!